pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionTest.java

```diff
@@ -971,4 +972,21 @@ public void testPendingAckMarkDeletePosition() throws Exception {
                     Integer.parseInt(lastConfirmedEntry[1]) - 2);
         });
     }
+
+    @Test
+    public void testTBLowWaterMark() throws Exception {
```

Comment:
Please also add an end to end test to make sure the client-side can handle the new exception well.

Reply:
We already have end to end test at `TransactionLowWaterMarkTest`.

Reply:
I have move the test from `TransactionTest` to  `TransactionLowWaterMarkTest`

Reply:
OK, the test is very much needed.